webpages/signup.php

```diff
@@ -1,157 +1,201 @@
+                }
+              else{
+            
+                $email = prepare($_POST["email"]);
+                    if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
+              $emailerr = "<div class=\"error\"> Invalid email format </div>";
+            }
+            else {
+              $emailerr="";
+              $flag1 = 1;
+            }
+            
+                }
+                if(empty($_POST["pass"]))
+                {
+                  $passerr = "<div class=\"error\">  * Password is required! </div>";
+                }
+              else{
+                $password = prepare($_POST["pass"]);
+                if (strlen((string)$password)<6)
+                {
+                  $passerr = "<div class=\"error\"> Your password must be at least 6 characters long.</div>";
+                }
+                else{
+                  $passerr="";
+                  $flag2 = 1;
+                }
+              }
+              if(empty($_POST["cpass"]))
+                {
+                  $pass2err = "<div class=\"error\">  * Confirmation is required! </div>";
+                }
+              else{
+                $password2 = prepare($_POST["cpass"]);
+                if ($password != $password2)
+                {
+                  $pass2err = "<div class=\"error\">Confirm Password do not match Password</div>";
+                }
+                else{
+                  $pass2err ="";
+                  $flag3 = 1;
+                } 
+                }
+            
+              
+                if(empty($_POST["fname"]))
+                {
+                  $fnameerr = "<div class=\"error\">  * First Name is required! </div>";
+                }
+              else{
+                $fname = prepare($_POST["fname"]);
+                if (preg_match("/[^a-zA-Z]/", $fname))
+                {
+                  $fnameerr = "<div class=\"error\"> First Name can't have special characters </div>";
+                }
+                else{
+                  $fnameerr ="";
+                  $flag4 =1;
+                }
+                }  
+                if(empty($_POST["lname"]))
+                {
+                  $lnameerr = "<div class=\"error\">  * Last Name is required! </div>";
+                }
+              else{
+                $lname = prepare($_POST["lname"]);
+                if (preg_match("/[^a-zA-Z]/", $lname))
+                {
+                  $fnameerr = "<div class=\"error\">  Last Name can't have special characters </div>";
+                }
+                else{
+                  $fnameerr ="";
+                  $flag5 =1;
+                }
+                }
+                $gender = prepare($_POST["gender"]);
+            
+                if ($flag1 === 1 && $flag2 === 1  && $flag3 ===1 && $flag4 === 1 && $flag5 === 1)
+                {
+                  $servername = $username = $passd = "";
+                  $servername = "localhost";
```

Comment:
create a file called config.php and store all the configuration variables in it only and import(require_once) it in your php file for use. Do not push this config file to github instead create a new file called sample.config.php and add all the variables to be filled in that only with dummy values so that the one who wants to use the app can simply cahnge the name of the file to config.php and add his/her credentials to it. Add config.php to a gitignore file(google what it does)